505.xml

```diff
@@ -0,0 +1,265 @@
+0 - CoAP (as defined in RFC 7252) with the additional support for block-wise transfer. CoAP is the default setting. 
+1 - CoAPS (as defined in RFC 7252) with the additional support for block-wise transfer 
+2 - HTTP 1.1 (as defined in RFC 7230) 
+3 - HTTPS 1.1 (as defined in RFC 7230) Additional values MAY be defined in the future. Any value not understood by the LwM2M Server MUST be ignored.]]></Description>
+			</Item>
+			<Item ID="11">
+				<Name>Profile Update Delivery Method</Name>
+				<Operations>R</Operations>
+				<MultipleInstances>Single</MultipleInstances>
+				<Mandatory>Mandatory</Mandatory>
+				<Type>Integer</Type>
+				<RangeEnumeration></RangeEnumeration>
+				<Units></Units>
+				<Description><![CDATA[The LwM2M Client uses this resource to
+indicate its support for transferring Profile
+images to the client either via the Package
+Resource (=push) or via the Package URI
+Resource (=pull) mechanism.
+0 - Pull only
+1 - Push only
+2 - Both. In this case the LwM2M Server
+MAY choose the preferred mechanism for
+conveying the profile image to the
+LwM2M Client.]]></Description>
+			</Item>
+			<Item ID="12">
+				<Name>Free Memory on SIM</Name>
+				<Operations>R</Operations>
+				<MultipleInstances>Single</MultipleInstances>
+				<Mandatory>Optional</Mandatory>
+				<Type>Integer</Type>
+				<RangeEnumeration></RangeEnumeration>
+				<Units>bit</Units>
+				<Description><![CDATA[Estimated current available amount of storage space on SIM which can store data and software in the LwM2M Device (expressed in kilobytes).]]></Description>
+			</Item>
+			<Item ID="13">
+				<Name>Total Memory on SIM</Name>
+				<Operations>R</Operations>
+				<MultipleInstances>Single</MultipleInstances>
+				<Mandatory>Optional</Mandatory>
+				<Type>Integer</Type>
+				<RangeEnumeration></RangeEnumeration>
+				<Units>bit</Units>
+				<Description><![CDATA[Total amount of storage space which can store data and software in the LwM2M Device]]></Description>
+			</Item>
+			<Item ID="14">
+				<Name>Integrated Circuit Card Identifier (ICCID)</Name>
+				<Operations>R</Operations>
+				<MultipleInstances>Multiple</MultipleInstances>
+				<Mandatory>Optional</Mandatory>
+				<Type>Integer</Type>
+				<RangeEnumeration></RangeEnumeration>
+				<Units></Units>
+				<Description><![CDATA[This EF provides a unique identification number for the UICC/Smart Cards. Please refer ETSI TS 102.22.1]]></Description>
+			</Item>
+			<Item ID="15">
+				<Name>eUICC ID</Name>
+				<Operations>R</Operations>
+				<MultipleInstances>Single</MultipleInstances>
+				<Mandatory>Optional</Mandatory>
+				<Type>String</Type>
+				<RangeEnumeration></RangeEnumeration>
+				<Units></Units>
+				<Description><![CDATA[eUICC-ID (a.k.a. EID), see GSMA SGP.02 and GSMA SGP.29 for definitions]]></Description>
+			</Item>
+			<Item ID="16">
```

Comment:
This is the profile type in consumer variant given on ES2+ I/F when indicating to SM-DP+ what type of profile to create. How is this information is encoded in the profile package such that it can be provided from the LwM2M client to the LwM2M server? Can you please provide information on the foreseen usage of this resource?
There is a reference [4] in the description of the resource. What is it referring to?

Reply:
It is not clear how the profile type information is provided to the device/eUICC such that a LwM2M server can request this information from the LwM2M client. This is not something specified in SGP.22 at least, but in TrustedConnectivityAlliance specification "eUICC Profile Package: Interoperable Format Technical Specification" there is an OPTIONAL field in ProfileHeader called "profileType" with the description "The "profileType" is a free optional text indicating for example, the name of the Profile issuer and the type of Profile.". As far as I can understand this string is not necessarily the same as the Profile Type in SGP.22. If it is the profileType field of ProfileHeader that is to be returned by this resource then it is better to refer to TCA spec rather then SGP.22, I think. Given an GSMA compliant eUICC I am not sure if/how this information is stored after profile decryption and installation and how you can obtain this information. I believe also nuSIM is reusing ProfileHeader structure from TCA spec, and maybe in nuSIM case it is possible to read out this field from nuSIM.

Reply:
We are just trying mirror what an SM-DP+ server would be providing. Do you think we need to call out the TCA spec to do that? Can we just say it is a "free optional text" field that is produced by the SM-DP+ server and consumed by the LwM2M client?

Reply:
According to SGP.21, Annex B: "The Operator defines its different Profile types (identified by a [non-standardised] Profile Description ID) which contains the Network Access Application like USIM,
file structure, data and applications, etc", so this is really operator specific. As we indicated, we are just duplicating what the current SM-DP+ server usage is.

Reply:
Indeed this is Operator specific information that is encrypted by the SM-DP+ and is decrypted in the eUICC/iUICC/nuSIM... Exactly what is written into this field seems to be SM-DP+ specific.
The LwM2M client may obtain this information from the eUICC/iUICC/nuSIM if an appropriate such interface exists which depends on the eUICC/iUICC/nuSIM realization (not sure that is specified in GSMA eSIM). "Free optional text produced by the SM-DP+" would be OK description.

Reply:
I changed the first sentence of the description to "Free optional text produced by the SM-DP+"